app/controllers/UpfrontPaymentController.scala

```diff
@@ -17,53 +17,78 @@
 package controllers
 
 import _root_.actions.Actions
-import controllers.UpfrontPaymentController.{upfrontPaymentAmountForm, upfrontPaymentForm}
-import models.{MockJourney, UserAnswers}
+import controllers.JourneyIncorrectStateRouter.logErrorAndRouteToDefaultPage
+import controllers.UpfrontPaymentController.upfrontPaymentAmountForm
+import essttp.journey.model.{Journey, Origins}
+import essttp.rootmodel.{AmountInPence, CanPayUpfront}
 import models.MoneyUtil.amountOfMoneyFormatter
-import essttp.rootmodel.AmountInPence
+import models.enumsforforms.CanPayUpfrontFormValue
+import models.forms.CanPayUpfrontForm
+import models.{MockJourney, UserAnswers}
+import play.api.data.Forms.mapping
 import play.api.data.{Form, Forms}
-import play.api.data.Forms.{mapping, nonEmptyText}
-import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import play.api.mvc._
+import requests.RequestSupport
+import services.JourneyService
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
 import util.Logging
-import views.html.{UpfrontPayment, UpfrontPaymentAmount, UpfrontSummary}
+import views.Views
 
 import javax.inject.{Inject, Singleton}
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
 class UpfrontPaymentController @Inject() (
-    as:                       Actions,
-    mcc:                      MessagesControllerComponents,
-    upfrontPaymentPage:       UpfrontPayment,
-    upfrontPaymentAmountPage: UpfrontPaymentAmount,
-    upfrontSummaryPage:       UpfrontSummary
+    as:             Actions,
+    mcc:            MessagesControllerComponents,
+    views:          Views,
+    journeyService: JourneyService,
+    requestSupport: RequestSupport
 )(implicit ec: ExecutionContext)
   extends FrontendController(mcc)
   with Logging {
 
-  val upfrontPayment: Action[AnyContent] = as.authenticatedJourneyAction.async { implicit request =>
-    Future.successful(Ok(upfrontPaymentPage(upfrontPaymentForm())))
+  import requestSupport._
+
+  val canYouMakeAnUpfrontPayment: Action[AnyContent] = as.authenticatedJourneyAction { implicit request =>
+    request.journey match {
+      case j: Journey.Stages.AfterStarted       => logErrorAndRouteToDefaultPage(j)
+      case j: Journey.Stages.AfterComputedTaxId => logErrorAndRouteToDefaultPage(j)
+      case j: Journey.HasEligibilityCheckResult => displayPage(j)
+    }
   }
 
-  val upfrontPaymentSubmit: Action[AnyContent] = as.authenticatedJourneyAction.async { implicit request =>
-    upfrontPaymentForm()
+  private def displayPage(journey: Journey.HasEligibilityCheckResult)(implicit request: Request[_]): Result = {
+    val backUrl = journey.origin match {
```

Comment:
wouldn't it be simpler to have something like this?
```scala
val back = routes.YourBillController.yourBill()
Ok(views.canYouMakeAnUpFrontPayment(CanPayUpfrontForm.form, Some(back.url)))
```

Reply:
oops sorry forgot that one will do now